crates/engine/tree/src/persistence.rs

```diff
@@ -203,6 +211,12 @@ impl PersistenceHandle {
         Self { sender }
     }
 
+    /// Sends a specific [`PersistenceAction`] in the contained channel. The caller is responsible
+    /// for creating any channels for the given action.
+    pub fn send_action(&self, action: PersistenceAction) {
+        self.sender.send(action).expect("should be able to send");
```

Comment:
Could we prevent this panic? If the caller should take care of creating the channels then this is open to failures from outside?

Reply:
Yeah, we could bubble the error